static/js/containers/App.js

```diff
@@ -1,38 +1,16 @@
 // @flow
 import React from 'react';
-import { connect } from 'react-redux';
-import {
-  updateCheckbox
-} from '../actions';
+import { Route } from 'react-router-dom';
 
-class App extends React.Component {
-  props: {
-    dispatch: () => void,
-    checkbox: {
-      checked: boolean,
-    },
-  };
-
-  handleClick(e) {
-    const { dispatch } = this.props;
-    dispatch(updateCheckbox(e.target.checked));
-  }
+import HomePage from './HomePage';
 
+export default class App extends React.Component {
   render() {
-    const { checked } = this.props.checkbox;
-    return <div>
-      <div id="app-body">
-        Click the checkbox:
-        <input type="checkbox" checked={checked} onClick={this.handleClick.bind(this)} />
+    const { match } = this.props;
+    return (
+      <div>
+        <Route exact path={match.url} component={HomePage}/>
```

Comment:
`match.url` allows for relative nesting without having to care what's above you. Maybe we don't need this for `App`, but it would allow for us to mount `App`  in MM like `<Route path="/discussions" component={discussions.App}/>` or something like that without having to modify `App`. I think that's generally a good approach for keeping these components reusable without assumptions of where they live relative to the root.